lib/machines.js

```diff
@@ -858,8 +854,20 @@ function getCreateOptions(req) {
             }
         });
 
+        opts.internal_metadata = credentials;
+    }
+
+
+    if (params.encrypted === true) {
+        if (!opts.internal_metadata) {
+            opts.internal_metadata = {};
+        }
+        opts.internal_metadata.encrypted = true;
+    }
+
+    if (opts.internal_metadata) {
         // Pass a string to VMAPI, not an object, or validation will fail
-        opts.internal_metadata = JSON.stringify(credentials);
+        opts.internal_metadata = JSON.stringify(opts.internal_metadata);
```

Comment:
Hmmm... this doesn't seem right. The `internal_metadata` property should be passed as an object.

Reply:
It's lame, but it's how it works: https://github.com/joyent/sdc-vmapi/blob/master/lib/common/validation.js#L1755-L1758

Some testing, with this CloudAPI's code. These are Job's parameters:
```
  "internal_metadata": {
    "root_pw": "secret",
    "admin_pw": "secret",
    "encrypted": true,
    "locality": "{\"far\":\"af4167f0-beda-4af9-9ae4-99d544499c14\",\"strict\":true}"
  },
```

And it of course went into an encrypted server.

Reply:
That code (linked above) is only for the vms.list endpoint - not for vms.create endpoint.

It looks like this is the code that does the string to object transformation for internal_metadata during vm.create:
https://github.com/joyent/sdc-vmapi/blob/35fbeaabd424ccf7b2d4c6d305789c5762b7b8b3/lib/common/validation.js#L2455

I don't know why one technique complains and the other does not - that seems like a validation bug (in vmapi) to me.